contracts/contracts/harvest/OETHBaseHarvester.sol

```diff
@@ -0,0 +1,206 @@
+        address indexed rewardToken,
+        address indexed swappedInto,
+        uint8 swapPlatform,
+        uint256 amountIn,
+        uint256 amountOut
+    );
+
+    /**
+     * @notice Verifies that the caller is either Governor or Strategist.
+     */
+    modifier onlyGovernorOrStrategist() {
+        require(
+            msg.sender == vault.strategistAddr() || isGovernor(),
+            "Caller is not the Strategist or Governor"
+        );
+        _;
+    }
+
+    constructor(
+        address _vault,
+        address _amoStrategy,
+        address _aero,
+        address _weth,
+        address _swapRouter
+    ) {
+        vault = IVault(_vault);
+        amoStrategy = IStrategy(_amoStrategy);
+        aero = IERC20(_aero);
+        weth = IERC20(_weth);
+        swapRouter = ISwapRouter(_swapRouter);
+    }
+
+    /**
+     * @notice Collects AERO from AMO strategy and
+     *      sends it to the Strategist multisig.
+     *      Anyone can call it.
+     */
+    function harvest() external {
+        address strategistAddr = vault.strategistAddr();
+        require(strategistAddr != address(0), "Guardian address not set");
+
+        // Collect all AERO
+        amoStrategy.collectRewardTokens();
+
+        uint256 aeroBalance = aero.balanceOf(address(this));
+        if (aeroBalance == 0) {
+            // Do nothing if there's no AERO to transfer
+            return;
+        }
+
+        // Transfer everything to Strategist
+        aero.safeTransfer(strategistAddr, aeroBalance);
+    }
+
+    /**
+     * @notice Harvests AERO from AMO strategy and then swaps some (or all)
+     *          of it into WETH to distribute yield and fee.
+     *         When `feeBps` is set to 10000 (100%), all WETH received is
+     *          sent to strategist.
+     *
+     * @param aeroToSwap Amount of AERO to swap
+     * @param minWETHExpected Min. amount of WETH to expect
+     * @param feeBps Performance fee bps (Sent to strategist)
+     * @param yieldRecipient Yield recipient (must be Vault or Dripper)
+     */
+    function harvestAndSwap(
+        uint256 aeroToSwap,
+        uint256 minWETHExpected,
+        uint256 feeBps,
+        address yieldRecipient
+    ) external onlyGovernorOrStrategist {
+        address strategistAddr = vault.strategistAddr();
+        require(strategistAddr != address(0), "Guardian address not set");
+
+        require(feeBps <= 10000, "Invalid Fee Bps");
+
+        // Collect all AERO
+        amoStrategy.collectRewardTokens();
+
+        uint256 aeroBalance = aero.balanceOf(address(this));
+        if (aeroBalance == 0) {
+            // Do nothing if there's no AERO to transfer/swap
+            return;
+        }
+
+        if (aeroToSwap > 0) {
+            if (aeroBalance < aeroToSwap) {
+                // Transfer in balance from the multisig as needed
+                // slither-disable-next-line unchecked-transfer arbitrary-send-erc20
+                aero.safeTransferFrom(
```

Comment:
I understand why you added this, but this seems a bit over-engineering to me. I would rather completely remove the public `harvest() `and keep only the ones that have the restricted call, to prevent front-running.
 
However, as I don't know what the project requirements are and this doesn't seem to be unsafe, so I have no problem keeping it like this.

Has this been considered to create a new role, only for simple harvest, and grant this role to a OZ Relayer?

Reply:
> Has this been considered to create a new role, only for simple harvest, and grant this role to a OZ Relayer?

Yep, that's an option. However, I didn't wanna complicate the storage layout. Sooner or later, we want the harvester to inherit from `AbstractHarvester` like on mainnet.

Thinking of it, I guess one option is to completely discard this contract and proxy when we plan to make that transition instead of trying to upgrade the implementation at that time

Reply:
This seems a good idea indeed!